evap/results/tests/test_tools.py

```diff
@@ -190,3 +192,106 @@ def test_result_calculation_with_no_contributor_rating_question_does_not_fail(se
+        contributor_all_contribution_visible_to, contributor_all_contribution_delegates = textanswers_visible_to(self.contributor_all_contribution)
+
+        users_seeing_general_contribution = (set(), set())
+        users_seeing_responsible1_contribution = (set(), set())
+        users_seeing_responsible2_contribution = (set(), set())
+        users_seeing_contributor_own_contribution = (set(), set())
+        users_seeing_contributor_course_contribution = (set(), set())
+        users_seeing_contributor_all_contribution = (set(), set())
+
+        for user in UserProfile.objects.all():
+            represented_users = [user] + list(user.represented_users.all())
+            if user_can_see_text_answer(user, represented_users, self.general_contribution_textanswer, 'full'):
+                if user_can_see_text_answer(user, [user], self.general_contribution_textanswer, 'full'):
+                    users_seeing_general_contribution[0].add(user)
+                else:
+                    users_seeing_general_contribution[1].add(user)
+            if user_can_see_text_answer(user, represented_users, self.responsible1_textanswer, 'full'):
+                if user_can_see_text_answer(user, [user], self.responsible1_textanswer, 'full'):
+                    users_seeing_responsible1_contribution[0].add(user)
+                else:
+                    users_seeing_responsible1_contribution[1].add(user)
+            if user_can_see_text_answer(user, represented_users, self.responsible2_textanswer, 'full'):
+                if user_can_see_text_answer(user, [user], self.responsible2_textanswer, 'full'):
+                    users_seeing_responsible2_contribution[0].add(user)
+                else:
+                    users_seeing_responsible2_contribution[1].add(user)
+            if user_can_see_text_answer(user, represented_users, self.contributor_own_textanswer, 'full'):
+                if user_can_see_text_answer(user, [user], self.contributor_own_textanswer, 'full'):
+                    users_seeing_contributor_own_contribution[0].add(user)
+                else:
+                    users_seeing_contributor_own_contribution[1].add(user)
+            if user_can_see_text_answer(user, represented_users, self.contributor_course_textanswer, 'full'):
+                if user_can_see_text_answer(user, [user], self.contributor_course_textanswer, 'full'):
+                    users_seeing_contributor_course_contribution[0].add(user)
+                else:
+                    users_seeing_contributor_course_contribution[1].add(user)
+            if user_can_see_text_answer(user, represented_users, self.contributor_all_textanswer, 'full'):
+                if user_can_see_text_answer(user, [user], self.contributor_all_textanswer, 'full'):
+                    users_seeing_contributor_all_contribution[0].add(user)
+                else:
+                    users_seeing_contributor_all_contribution[1].add(user)
```

Comment:
can we have something... more generic? i mean we have six contributions, each of those has a comment, the result of `textanswers_visible_to`, and then a bunch of users who can or can not see it. i have the feeling all this should be possible with a bunch of loops.